source/extensions/load_balancing_policies/override_host/selected_hosts.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "source/common/protobuf/protobuf.h"
+
+#include "absl/status/statusor.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace LoadBalancingPolices {
+namespace OverrideHost {
+
+// The SelectedHosts struct is used to represent parsed proto or headers with
+// the pre-selected primary and retry hosts.
+// The structure is immutable and can only be created via the factory methods.
+struct SelectedHosts {
+  struct Endpoint {
+  public:
+    struct Address {
+      const std::string address;
+      const uint32_t port;
+    };
+
+    const Address address;
+  };
+
+  const Endpoint primary;
+  const std::vector<Endpoint> failover;
```

Comment:
It's unecessary to split the primary address and fallback addreses into two field. We will iterate then in order anyway. And it's also unnecessary to use the `Address` struct.

A simple `std::vector<std::string>` could be used as the `SelectedHosts`